File: step_definitions/Verify Accessibility and Labeling of Filters on Desktop and Mobile.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import com.page_objects.FilterPage;

public class FilterSteps extends FilterPage {

    private TestHarness testHarness = new TestHarness();

    @Given("I open the application in a desktop browser")
    public void iOpenTheApplicationInADesktopBrowser() {
        launchApplicationOnDesktop();
    }

    @Then("the application should load successfully")
    public void theApplicationShouldLoadSuccessfully() {
        verifyApplicationLoad();
    }

    @Given("I navigate to the product listing page on a desktop browser")
    public void iNavigateToTheProductListingPageOnADesktopBrowser() {
        navigateToProductListingPageOnDesktop();
    }

    @Then("the product listing page should display correctly")
    public void theProductListingPageShouldDisplayCorrectly() {
        verifyProductListingPageDisplay();
    }

    @Given("I locate the filter section on the desktop interface")
    public void iLocateTheFilterSectionOnTheDesktopInterface() {
        locateFilterSectionOnDesktop();
    }

    @Then("filters should be visible and clearly labeled")
    public void filtersShouldBeVisibleAndClearlyLabeled() {
        verifyFiltersVisibilityAndLabeling();
    }

    @Given("I access the application on a mobile browser")
    public void iAccessTheApplicationOnAMobileBrowser() {
        launchApplicationOnMobile();
    }

    @Then("the application should load successfully on mobile")
    public void theApplicationShouldLoadSuccessfullyOnMobile() {
        verifyApplicationLoadOnMobile();
    }

    @Given("I navigate to the product listing page on mobile")
    public void iNavigateToTheProductListingPageOnMobile() {
        navigateToProductListingPageOnMobile();
    }

    @Then("the product listing page should display correctly on mobile")
    public void theProductListingPageShouldDisplayCorrectlyOnMobile() {
        verifyProductListingPageDisplayOnMobile();
    }

    @Given("I locate the filter section on the mobile interface")
    public void iLocateTheFilterSectionOnTheMobileInterface() {
        locateFilterSectionOnMobile();
    }

    @Then("filters should be visible and clearly labeled on mobile")
    public void filtersShouldBeVisibleAndClearlyLabeledOnMobile() {
        verifyFiltersVisibilityAndLabelingOnMobile();
    }

    @Given("I access the filter section")
    public void iAccessTheFilterSection() {
        accessFilterSection();
    }

    @Then("filters should be accessible with one click or tap")
    public void filtersShouldBeAccessibleWithOneClickOrTap() {
        verifyFiltersAccessibility();
    }

    @Given("I resize the browser window on desktop")
    public void iResizeTheBrowserWindowOnDesktop() {
        resizeBrowserWindowOnDesktop();
    }

    @Then("filter labels should adjust correctly to different screen sizes")
    public void filterLabelsShouldAdjustCorrectlyToDifferentScreenSizes() {
        verifyFilterLabelsResponsivenessOnDesktop();
    }

    @Given("I view the filter labels on different mobile screen sizes")
    public void iViewTheFilterLabelsOnDifferentMobileScreenSizes() {
        viewFilterLabelsOnDifferentMobileScreenSizes();
    }

    @Then("filter labels should adjust correctly on different mobile screen sizes")
    public void filterLabelsShouldAdjustCorrectlyOnDifferentMobileScreenSizes() {
        verifyFilterLabelsResponsivenessOnMobile();
    }

    @Given("I use keyboard shortcuts to access filters on desktop")
    public void iUseKeyboardShortcutsToAccessFiltersOnDesktop() {
        useKeyboardShortcutsForFiltersOnDesktop();
    }

    @Then("filters should be accessible using keyboard shortcuts")
    public void filtersShouldBeAccessibleUsingKeyboardShortcuts() {
        verifyFiltersAccessibilityWithKeyboard();
    }

    @Given("I use voice commands to access filters on mobile")
    public void iUseVoiceCommandsToAccessFiltersOnMobile() {
        useVoiceCommandsForFiltersOnMobile();
    }

    @Then("filters should be accessible using voice commands")
    public void filtersShouldBeAccessibleUsingVoiceCommands() {
        verifyFiltersAccessibilityWithVoiceCommands();
    }

    @Given("I view the filter labels")
    public void iViewTheFilterLabels() {
        viewFilterLabels();
    }

    @Then("filter labels should not overlap with other UI elements")
    public void filterLabelsShouldNotOverlapWithOtherUIElements() {
        verifyFilterLabelsNoOverlap();
    }

    @Given("I check the filter labels in different lighting conditions on mobile")
    public void iCheckTheFilterLabelsInDifferentLightingConditionsOnMobile() {
        checkFilterLabelsInDifferentLightingConditionsOnMobile();
    }

    @Then("filter labels should be readable in different lighting conditions")
    public void filterLabelsShouldBeReadableInDifferentLightingConditions() {
        verifyFilterLabelsReadabilityInLightingConditions();
    }

    @Given("I apply filters on both platforms")
    public void iApplyFiltersOnBothPlatforms() {
        applyFiltersOnBothPlatforms();
    }

    @Then("filters should be applied correctly on both desktop and mobile")
    public void filtersShouldBeAppliedCorrectlyOnBothDesktopAndMobile() {
        verifyFiltersAppliedCorrectly();
    }

    @Given("I reset the filters after application")
    public void iResetTheFiltersAfterApplication() {
        resetFiltersAfterApplication();
    }

    @Then("filters should reset correctly after application")
    public void filtersShouldResetCorrectlyAfterApplication() {
        verifyFiltersResetCorrectly();
    }
}

File: step_definitions/Verify Accessibility and Labeling of Filters on Desktop and Mobile.js
import { Given, Then } from "cypress-cucumber-preprocessor/steps";
import FilterPage from "../pages/filterPage";

Given("I open the application in a desktop browser", () => {
 FilterPage.visitDesktop();
});

Then("the application should load successfully", () => {
 FilterPage.verifyAppLoad();
});

Given("I navigate to the product listing page on a desktop browser", () => {
 FilterPage.navigateToProductListingDesktop();
});

Then("the product listing page should display correctly", () => {
 FilterPage.verifyProductListingDisplay();
});

Given("I locate the filter section on the desktop interface", () => {
 FilterPage.locateFilterSectionDesktop();
});

Then("filters should be visible and clearly labeled", () => {
 FilterPage.verifyFiltersVisibilityAndLabeling();
});

Given("I access the application on a mobile browser", () => {
 FilterPage.visitMobile();
});

Then("the application should load successfully on mobile", () => {
 FilterPage.verifyAppLoadMobile();
});

Given("I navigate to the product listing page on mobile", () => {
 FilterPage.navigateToProductListingMobile();
});

Then("the product listing page should display correctly on mobile", () => {
 FilterPage.verifyProductListingDisplayMobile();
});

Given("I locate the filter section on the mobile interface", () => {
 FilterPage.locateFilterSectionMobile();
});

Then("filters should be visible and clearly labeled on mobile", () => {
 FilterPage.verifyFiltersVisibilityAndLabelingMobile();
});

Given("I access the filter section", () => {
 FilterPage.accessFilterSection();
});

Then("filters should be accessible with one click or tap", () => {
 FilterPage.verifyFiltersAccessibility();
});

Given("I resize the browser window on desktop", () => {
 FilterPage.resizeBrowserWindowDesktop();
});

Then("filter labels should adjust correctly to different screen sizes", () => {
 FilterPage.verifyFilterLabelsResponsivenessDesktop();
});

Given("I view the filter labels on different mobile screen sizes", () => {
 FilterPage.viewFilterLabelsMobileScreens();
});

Then("filter labels should adjust correctly on different mobile screen sizes", () => {
 FilterPage.verifyFilterLabelsResponsivenessMobile();
});

Given("I use keyboard shortcuts to access filters on desktop", () => {
 FilterPage.useKeyboardShortcutsDesktop();
});

Then("filters should be accessible using keyboard shortcuts", () => {
 FilterPage.verifyKeyboardShortcutsAccessibility();
});

Given("I use voice commands to access filters on mobile", () => {
 FilterPage.useVoiceCommandsMobile();
});

Then("filters should be accessible using voice commands", () => {
 FilterPage.verifyVoiceCommandsAccessibility();
});

Given("I view the filter labels", () => {
 FilterPage.viewFilterLabels();
});

Then("filter labels should not overlap with other UI elements", () => {
 FilterPage.verifyNoOverlapWithUIElements();
});

Given("I check the filter labels in different lighting conditions on mobile", () => {
 FilterPage.checkFilterLabelsLightingConditionsMobile();
});

Then("filter labels should be readable in different lighting conditions", () => {
 FilterPage.verifyFilterLabelsReadabilityLightingConditionsMobile();
});

Given("I apply filters on both platforms", () => {
 FilterPage.applyFiltersBothPlatforms();
});

Then("filters should be applied correctly on both desktop and mobile", () => {
 FilterPage.verifyFiltersApplicationBothPlatforms();
});

Given("I reset the filters after application", () => {
 FilterPage.resetFiltersAfterApplication();
});

Then("filters should reset correctly after application", () => {
 FilterPage.verifyFiltersReset();
});